x-pack/plugins/apm/public/services/rest/apm/metrics.ts

```diff
@@ -4,27 +4,30 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { MetricsChartAPIResponse } from '../../../../server/lib/metrics/get_all_metrics_chart_data';
+import { MetricsChartsByAgentAPIResponse } from '../../../../server/lib/metrics/get_metrics_chart_data_by_agent';
 import { callApi } from '../callApi';
 import { getEncodedEsQuery } from './apm';
 
-export async function loadMetricsChartDataForService({
+export async function loadMetricsChartData({
   serviceName,
+  agentName,
   start,
   end,
   kuery
 }: {
   serviceName: string;
+  agentName?: string;
   start: string;
   end: string;
-  kuery: string | undefined;
+  kuery?: string;
```

Comment:
So we're trying to make a distinction between:
```
{ kuery: undefined }
```
and
```
{}
```
?

This seems like a confusing use of `undefined` to me but I'll change it however you want it.

Reply:
We are trying to make a distinction between:
```ts
loadMetricsChartData({ serviceName, start, end }) // should fail due to missing kuery
```

and

```ts
loadMetricsChartData({ serviceName, start, end, kuery }) // should not fail
```

Reply:
But in the second example why can `kuery` be `undefined`? And if it can be `undefined`, how is it different from the first example that you said should fail? Sorry, that's just the part that's really unclear about this `| undefined` syntax to me.

Reply:
I changed it back, just trying to understand what's going on with this. :)

Reply:
> But in the second example why can kuery be undefined? And if it can be undefined, how is it different from the first example that you said should fail? 

Yes, in both first and second example the value of `kuery` can be undefined. The thing I'm trying to ensure, is that the developer always passes in `kuery`. I noticed we had this error, where `kuery` was not passed to the rest methods, which meant that those components did not take `kuery` into account - this was a bug from the end-user point of view.

This is a an example with `string | undefined`. It will require `name` to be parsed in:
https://www.typescriptlang.org/play/#src=function%20hello(name%3A%20string%20%7C%20undefined)%20%7B%0D%0A%20%20return%20true%3B%0D%0A%7D%0D%0A%0D%0Aconst%20name1%20%3D%20'John%20Doe'%3B%0D%0Ahello(name)%3B%20%2F%2F%20%E2%9C%85%20No%20error%0D%0A%0D%0Aconst%20name2%20%3D%20undefined%3B%20%0D%0Ahello(name2)%3B%20%2F%2F%20%E2%9C%85%20No%20error%20%0D%0A%0D%0Ahello()%3B%20%2F%2F%20%E2%9C%85%20Error%20(which%20it%20should)

This is a example witjh `?string`. It will not require `name` to be passed in. **So the developer could easily forget - which would be a mistake.**
https://www.typescriptlang.org/play/#src=function%20hello(name%3F%3A%20string)%20%7B%0D%0A%20%20return%20true%3B%0D%0A%7D%0D%0A%0D%0Aconst%20name1%20%3D%20'John%20Doe'%3B%0D%0Ahello(name)%3B%20%2F%2F%20%E2%9C%85%20No%20error%0D%0A%0D%0Aconst%20name2%20%3D%20undefined%3B%20%0D%0Ahello(name2)%3B%20%2F%2F%20%E2%9C%85%20No%20error%20%0D%0A%0D%0Ahello()%3B%20%2F%2F%20%E2%9D%8C%20No%20error%20(which%20is%20a%20problem)

Reply:
Maybe as a follow up question: if we use `kuery?: string` how do you make sure that developers will not forget to call that method with `kuery` argument?